### ISSUE TITLE
Add tests

### ISSUE THREAD

**Issue by jaydrogers** [serversideup/docker-php#10]
# Summary
- It would be helpful to add tests

# Order
It would be nice to have each image go through these steps:
1. Setup (Pull base images & login to Docker)
2. Build
3. Test
4. Push

# What needs to be figured out yet
* What is the best way to "test" that PHP is working?
* Will it change on each deployment type (CLI, FPM, FPM-NGINX)

**Comment by szepeviktor** [comment]
> What is the best way to "test" that PHP is working?

Run Laravel unit tests??

**Comment by AkenRoberts** [comment]
> What is the best way to "test" that PHP is working?

I would think just PHP commands would be sufficient, e.g. `php -v`, `php -i`, etc. They'll also let you assert that extensions are installed.

**Comment by jaydrogers** [comment]
Awesome! Thanks for the insight @AkenRoberts!

**Comment by szepeviktor** [comment]
> I would think just PHP commands would be sufficient, e.g. `php -v`, `php -i`, etc.

That is like assuming an SQL server is okay by querying its version.
Real life **functional** tests reveal real life problems.

**Comment by AkenRoberts** [comment]
Right. Maybe more functional tests to ensure PHP is behaving would be useful for a container providing it. I don't see that immediately (_'certainly PHP itself is well-tested'_), but admittedly that could be naive.

Definitely choose the level of testing that ensures the value provided to end users is maintained. 👍🏼
